modules/element-manager/assets/js/app.js

```diff
@@ -0,0 +1,527 @@
+		isSaving: false,
+		isUnsavedChanges: false,
+	} );
+	const [ isConfirmDialogOpen, setIsConfirmDialogOpen ] = useState( false );
+	const [ isSnackbarOpen, setIsSnackbarOpen ] = useState( false );
+
+	const getUsageWidget = ( widgetName ) => {
+		if ( ! usageWidgets.data || ! usageWidgets.data.hasOwnProperty( widgetName ) ) {
+			return 0;
+		}
+
+		return usageWidgets.data[ widgetName ];
+	}
+
+	const sortedAndFilteredWidgets = useMemo( () => {
+		let filteredWidgets = widgets.filter( ( widget ) => {
+			return widget.title.toLowerCase().includes( searchKeyword.toLowerCase() );
+		} );
+
+		if ( '' !== filterByPlugin ) {
+			filteredWidgets = filteredWidgets.filter( ( widget ) => {
+				return widget.plugin.toLowerCase() === filterByPlugin.toLowerCase();
+			} );
+		}
+
+		filteredWidgets.sort( ( a, b ) => {
+			let aValue;
+			let bValue;
+
+			if ( 'widget' === sortingColumn ) {
+				aValue = a.title;
+				bValue = b.title;
+			}
+
+			if ( 'usage' === sortingColumn ) {
+				aValue = getUsageWidget( a.name );
+				bValue = getUsageWidget( b.name );
+			}
+
+			if ( aValue === bValue ) {
+				return 0;
+			}
+
+			if ( 'asc' === sortingDirection ) {
+				return aValue < bValue ? -1 : 1;
+			}
+
+			return aValue > bValue ? -1 : 1;
+		} );
+
+		return filteredWidgets;
+	}, [ widgets, searchKeyword, sortingColumn, sortingDirection, filterByPlugin, usageWidgets ] );
+
+	const getSortingIndicatorClasses = ( column ) => {
+		if ( sortingColumn !== column ) {
+			return '';
+		}
+
+		if ( 'asc' === sortingDirection ) {
+			return 'sorted asc';
+		}
+
+		return 'sorted desc';
+	};
+
+	const onSortingClicked = ( column ) => {
+		if ( sortingColumn === column ) {
+			if ( 'asc' === sortingDirection ) {
+				setSortingDirection( 'desc' );
+			} else {
+				setSortingDirection( 'asc' );
+			}
+		} else {
+			setSortingColumn( column );
+			setSortingDirection( 'asc' );
+		}
+	};
+
+	const onSaveClicked = async () => {
+		setIsConfirmDialogOpen( false );
+		setChangeProgress( { ...changeProgress, isSaving: true } );
+
+		await saveDisabledWidgets( widgetsDisabled );
+
+		setChangeProgress( { ...changeProgress, isSaving: false, isUnsavedChanges: false } );
+
+		setIsSnackbarOpen( true );
+	};
+
+	const deactivateAllUnusedWidgets = () => {
+		const widgetsToDeactivate = widgets.filter( ( widget ) => {
+			return ! usageWidgets.data.hasOwnProperty( widget.name ) || widgetsDisabled.includes( widget.name );
+		} );
+
+		setWidgetsDisabled( widgetsToDeactivate.map( ( widget ) => {
+			return widget.name;
+		} ) );
+	};
+
+	const enableAllWidgets = () => {
+		setWidgetsDisabled( [] );
+	};
+
+	const onScanUsageElementsClicked = async () => {
+		setUsageWidgets( { ...usageWidgets, isLoading: true } );
+
+		const data = await getUsageWidgets();
+
+		setUsageWidgets( { data, isLoading: false } );
+		setSortingColumn( 'usage' );
+		setSortingDirection( 'desc' );
+	};
+
+	const UsageTimesColumn = ( { widgetName } ) => {
```

Comment:
it's a bad practice (and hurts performance) to create components inside components